src/libraries/System.Linq/src/System/Linq/Chunk.cs

```diff
@@ -50,27 +51,24 @@ public static IEnumerable<TSource[]> Chunk<TSource>(this IEnumerable<TSource> so
         private static IEnumerable<TSource[]> ChunkIterator<TSource>(IEnumerable<TSource> source, int size)
         {
             using IEnumerator<TSource> e = source.GetEnumerator();
-            while (e.MoveNext())
-            {
-                TSource[] chunk = new TSource[size];
-                chunk[0] = e.Current;
 
-                int i = 1;
-                for (; i < chunk.Length && e.MoveNext(); i++)
+            if (e.MoveNext())
+            {
+                List<TSource> chunkBuilder = new();
+                do
                 {
-                    chunk[i] = e.Current;
-                }
+                    chunkBuilder.Clear();
```

Comment:
Redundant on the first call, I would move this to the bottom of the loop body.

Reply:
I had done it this way for brevity originally, but moving it to the bottom only adds a couple of lines and is more intuitive.